sigstore/dsse.py

```diff
@@ -256,6 +289,19 @@ def _sign(key: ec.EllipticCurvePrivateKey, stmt: Statement) -> Envelope:
     )
 
 
+def _sign_payload(
```

Comment:
Rather than a new `_sign_payload` function, can we make `_sign` take `Statement | RawPayload`? That'll require some internal checks but will be a bit shorter 🙂